engine/class_modules/paladin/sc_paladin_holy.cpp

```diff
@@ -450,16 +455,90 @@ struct lights_hammer_t : public paladin_spell_t
 
 // Light of Dawn (Holy) =======================================================
 
-struct light_of_dawn_t : public paladin_heal_t
+struct light_of_dawn_t : public holy_power_consumer_t<paladin_heal_t>
 {
   light_of_dawn_t( paladin_t* p, const std::string& options_str )
-    : paladin_heal_t( "light_of_dawn", p, p->find_class_spell( "Light of Dawn" ) )
+    : holy_power_consumer_t( "light_of_dawn", p, p->find_class_spell( "Light of Dawn" ) )
   {
     parse_options( options_str );
 
     aoe = 6;
+  }
+
+  void execute() override
+  {
+    holy_power_consumer_t::execute();
+
+    // deal with awakening
+    if ( p()->talents.awakening->ok() )
+    {
+      if ( rng().roll( p()->talents.awakening->effectN( 1 ).percent() ) )
+      {
+        buff_t* main_buff           = p()->buffs.avenging_wrath;
+        timespan_t trigger_duration = timespan_t::from_seconds( p()->talents.awakening->effectN( 2 ).base_value() );
+        if ( main_buff->check() )
+        {
+          p()->buffs.avengers_might->extend_duration( p(), trigger_duration );
+        }
+        else
+        {
+          main_buff->trigger( 1, buff_t::DEFAULT_VALUE(), -1.0, trigger_duration );
+        }
+      }
+    }
+  }
+};
+
+// Word Of Glory (Holy) =======================================================
+
+struct word_of_glory_t : public holy_power_consumer_t<paladin_heal_t>
+{
+  word_of_glory_t( paladin_t* p, const std::string& options_str )
+    : holy_power_consumer_t( "word_of_glory", p, p->find_class_spell( "Word Of Glory" ) )
+  {
+    parse_options( options_str );
+  }
+
+  void execute() override
+  {
+    holy_power_consumer_t::execute();
 
-    cooldown = p->cooldowns.light_of_dawn;
+    // deal with awakening
+    if ( p()->talents.awakening->ok() )
+    {
+      if ( rng().roll( p()->talents.awakening->effectN( 1 ).percent() ) )
+      {
+        buff_t* main_buff           = p()->buffs.avenging_wrath;
+        timespan_t trigger_duration = timespan_t::from_seconds( p()->talents.awakening->effectN( 2 ).base_value() );
+        if ( main_buff->check() )
+        {
+          p()->buffs.avengers_might->extend_duration( p(), trigger_duration );
+        }
+        else
+        {
+          main_buff->trigger( 1, buff_t::DEFAULT_VALUE(), -1.0, trigger_duration );
+        }
+      }
+    }
+  }
+};
+
+struct avenging_crusader_t : public paladin_spell_t
+{
+  avenging_crusader_t( paladin_t* p, const std::string& options_str )
+    : paladin_spell_t( "avenging_crusader", p, p->talents.avenging_crusader )
+  {
+    parse_options( options_str );
+
+    if ( !( p->talents.avenging_crusader->ok() ) )
+      background = true;
```

Comment:
This is automatically set during spelldata parsing

Reply:
Good to know